kanidmd_web_ui/src/components/admin_accounts.rs

```diff
@@ -0,0 +1,258 @@
+use crate::components::adminmenu::{Entity, GetError};
+use crate::components::alpha_warning_banner;
+use crate::constants::{CSS_CELL, CSS_PAGE_HEADER, CSS_TABLE};
+use crate::models;
+use crate::utils::{do_alert_error, init_request};
+use gloo::console;
+use std::collections::BTreeMap;
+// use yew::prelude::*;
+use yew::{html, Component, Context, Html, Properties};
+
+impl From<GetError> for AdminListAccountsMsg {
+    fn from(ge: GetError) -> Self {
+        AdminListAccountsMsg::Failed {
+            emsg: ge.err,
+            kopid: None,
+        }
+    }
+}
+
+pub struct AdminListAccounts {
+    state: ViewState,
+}
+
+// callback messaging for this confused pile of crab-bait
+pub enum AdminListAccountsMsg {
+    /// When the server responds and we need to update the page
+    Responded {
+        response: BTreeMap<String, Entity>,
+    },
+    Failed {
+        emsg: String,
+        kopid: Option<String>,
+    },
+}
+
+enum ViewState {
+    /// waiting for the page to load
+    Loading,
+    /// server has responded
+    Responded { response: BTreeMap<String, Entity> },
+    /// failed to pull the details
+    #[allow(dead_code)]
+    Failed {
+        // TODO: use this
+        emsg: String,
+        kopid: Option<String>,
+    },
+    #[allow(dead_code)]
+    /// Not authorized to pull the details
+    NotAuthorized {}, // TODO: use this
+}
+
+#[derive(PartialEq, Properties, Eq)]
+pub struct AdminListAccountsProps {
+    // for filtering and pagination
+    // #[allow(dead_code)]
+    // search: Option<String>,
+    // #[allow(dead_code)]
+    // page: Option<u32>,
+}
+
+/// Pulls all accounts (service or person-class) from the backend and returns a HashMap
+/// with the "name" field being the keys, for easy human-facing sortability.
+pub async fn get_accounts(token: &str) -> Result<AdminListAccountsMsg, GetError> {
+    // TODO: the actual pulling and turning into a BTreeMap in this and get_groups could *probably* be rolled up into one function? The result object differs but all the widgets are the same.
+    let mut all_accounts = BTreeMap::new();
+
+    // we iterate over these endpoints
+    let endpoints = [
```

Comment:
I think there is still an /v1/account endpoint that does this for you.

Reply:
Nope, 'tis gone. 😄 This allows me to really easily go "this is an x account type" too.

Reply:
Yep, you're right :) 

Well really they should be two seperate parts of admin ui anyway, a person admin and a robot/service account admin rather than "just accounts".

Reply:
🤔 I might argue that at the "list accounts" level:

- an account's an account - they've got usernames and auth methods and go into groups etc
- since they can (technically) move between `Service Account` <> `Person`, their root classification makes them similar enough to list them in one place
- having to iterate through two places for "something that authenticates through Kanidm" would be a little annoying for me as an admin

Obviously this is real early days for everything, but the difference between the two is also indicated visually:

<img width="662" alt="Screen Shot 2022-09-18 at 13 05 41" src="https://user-images.githubusercontent.com/168188/190883877-56a2d3ff-f0a8-4acd-912f-e745b7426901.png">